src/main/java/com/google/sps/servlets/NewsServlet.java

```diff
@@ -10,42 +10,132 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import com.google.sps.data.Article;
+import java.net.URL;
+import java.net.HttpURLConnection;
+import java.util.Scanner;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonElement;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * Servlet for generating news articles from keywords.
  */
 @WebServlet("/news")
 public class NewsServlet extends HttpServlet {
-  private List<String> keywords = new ArrayList<>();
-  
-  // Hardcoded list of articles. 
-  // In future implementation, will be generated based on key words.
-  private List<Article> articles = new ArrayList<>(
-      List.of(
-        new Article.Builder(
-            "Bolton says Trump turned a blind eye to the coronavirus pandemic",
-            "https://www.cnn.com/2020/06/24/politics/john-bolton-interview-cnntv/index.html")
-          .withWriter("Zachary Cohen")
-          .withDate("06/24/2020")
-          .build(),
-        new Article.Builder(
-            "Hearing goes off the rails when lawmaker keeps banging table",
-            "https://www.cnn.com/videos/politics/2020/06/24/louie-gohmert-bangs-table-judiciary-hearing-vpx.cnn")
-          .withDate("06/24/2020")
-          .build()
-      )   
+
+  private final String API_KEY = "test"; // Insert actual API key when testing.
+  private int numArticlesPerKeyword = 3;
+  private List<String> keywords = new ArrayList<String>(
+    List.of("Black Lives Matter", "COVID-19")
   );
+  private HashMap<String, List<Article>> articleMap = new HashMap<String, List<Article>>();
```

Comment:
[Nit] The last `<String, List<Article>>` definition is unnecessary. You can replace it with simply `<>`

For example: `private List<String> list = new ArrayList<>();`